python/ray/util/client/ray_client_helpers.py

```diff
@@ -41,3 +42,20 @@ def ray_connect_handler(job_config=None):
         ray._inside_client_test = False
         ray.disconnect()
         server.stop(0)
+
+
+class RayClientSerializationContext:
+    # NOTE(simon): Used for registering custom serializers. We cannot directly
+    # use the SerializationContext because it requires Ray workers. Please
+    # make sure to keep the API consistent.
+
+    def _unregister_cloudpickle_reducer(self, cls):
```

Comment:
Hmm, do we need any of the other methods for this? 

I think this approach is fine for now, but I wonder if a better long term goal to have a common base SerializationContext class and then have a more specialized one for workers.